AdSecGHTests/Parameters/AdSecSectionTests.cs

```diff
@@ -19,64 +19,59 @@
 namespace AdSecGHTests.Parameters {
   [Collection("GrasshopperFixture collection")]
   public class AdSecSectionTests {
-    [Fact(Skip = "Not Implemented Yet")]
+    [Fact]
     public void SerialiseUnflattenedSectionTest() {
-      ISection section = CreateSection();
+      var section = CreateSection();
 
-      IDesignCode designCode = EN1992.Part1_1.Edition_2004.NationalAnnex.DE.Edition_2013;
+      var designCode = EN1992.Part1_1.Edition_2004.NationalAnnex.DE.Edition_2013;
       var adSec = IAdSec.Create(designCode);
-      ISection flattened = adSec.Flatten(section);
+      var flattened = adSec.Flatten(section);
 
-      string fileName = Path.GetTempPath() + "AdSecSectionTest.ads";
+      string fileName = Path.GetTempPath() + "AdSecSectionTest-Unflattened.ads";
       File.WriteAllText(fileName, CreateJson(designCode, section));
 
       string json = File.ReadAllText(fileName);
-      ParsedResult jsonParser = JsonParser.Deserialize(json);
-      ISection actualSection = jsonParser.Sections[0];
+      var jsonParser = JsonParser.Deserialize(json);
+      var actualSection = jsonParser.Sections[0];
 
-      var expectedProfile = (IPerimeterProfile)flattened.Profile;
-      var actualProfile = (IPerimeterProfile)actualSection.Profile;
-
-      TestSection(flattened, actualSection);
+      TestSection(flattened, actualSection, true);
     }
 
     [Fact]
     public void SerialiaseFlattenedSectionTest() {
-      ISection section = CreateSection();
+      var section = CreateSection();
 
-      IDesignCode designCode = EN1992.Part1_1.Edition_2004.NationalAnnex.DE.Edition_2013;
+      var designCode = EN1992.Part1_1.Edition_2004.NationalAnnex.DE.Edition_2013;
       var adSec = IAdSec.Create(designCode);
-      ISection flattened = adSec.Flatten(section);
+      var flattened = adSec.Flatten(section);
 
       string fileName = Path.GetTempPath() + "AdSecSectionTest.ads";
       File.WriteAllText(fileName, CreateJson(designCode, flattened));
 
       string json = File.ReadAllText(fileName);
-      ParsedResult jsonParser = JsonParser.Deserialize(json);
-      ISection actualSection = jsonParser.Sections[0];
-
-      var expectedProfile = (IPerimeterProfile)flattened.Profile;
-      var actualProfile = (IPerimeterProfile)actualSection.Profile;
+      var jsonParser = JsonParser.Deserialize(json);
+      var actualSection = jsonParser.Sections[0];
 
       TestSection(flattened, actualSection);
     }
 
-    private void TestSection(ISection expected, ISection actual) {
+    private static void TestSection(ISection expected, ISection actual, bool unflattened = false) {
```

Comment:
can we put this comparison at the core GSA.GH project withi this signature
```csharp
public class SectionHelper
{
    public static bool Equal(ISection sectionA, ISection sectionB, bool unflattened){...}
}
```

then this method can simply call the previous one

```csharp
{
    Assert.True(SectionHelper.Equal(expected, actual, unflattened))
}
```